mLRS/Common/hal/esp/rx-hal-generic-2400-pa-td-esp32.h

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) MLRS project
+// GPL3
+// https://www.gnu.org/licenses/gpl-3.0.de.html
+// OlliW @ www.olliw.eu
+//*******************************************************
+// hal
+//********************************************************
+
+//-------------------------------------------------------
+// GENERIC 2400 True Diversity PA RX
+//-------------------------------------------------------
+
+#define DEVICE_HAS_SINGLE_LED
+//#define DEVICE_HAS_SERIAL_OR_DEBUG
+#define DEVICE_HAS_NO_DEBUG
+
+//-- Timers, Timing, EEPROM, and such stuff
+
+#define EE_START_PAGE             0 // 128 kB flash, 2 kB page
+
+//-- UARTS
+// UARTB = serial port
+// UART = output port, SBus or whatever
+// UARTC = debug port
+
+#define UARTB_USE_SERIAL
+#define UARTB_BAUD                RX_SERIAL_BAUDRATE
+#define UARTB_TXBUFSIZE           RX_SERIAL_TXBUFSIZE
+#define UARTB_RXBUFSIZE           RX_SERIAL_RXBUFSIZE
+
+#define UARTC_USE_SERIAL
+#define UARTC_BAUD                115200
+
+
+//-- SX1: SX12xx & SPI
+#define SPI_CS_IO                 27
+#define SPI_MISO                  33
+#define SPI_MOSI                  32
+#define SPI_SCK                   25
+#define SPI_FREQUENCY             16000000L
+
+#define SX_BUSY                   36
+#define SX_DIO1                   37
+#define SX_RESET                  26
+
+#define SX_RX_EN                  10
+#define SX_TX_EN                  14
+
+//-- SX2: SX12xx & SPI
+
+#define SX2_CS_IO                 13
+
+#define SX2_BUSY                  39
+#define SX2_DIO1                  34
+#define SX2_RESET                 21
+
+#define SX2_RX_EN                  9
+#define SX2_TX_EN                 15
+
+
+IRQHANDLER(void SX_DIO_EXTI_IRQHandler(void);)
+
+void sx_init_gpio(void)
+{
+    gpio_init(SX_DIO1, IO_MODE_INPUT_ANALOG);
+    gpio_init(SX_BUSY, IO_MODE_INPUT_PU);
+    gpio_init(SX_TX_EN, IO_MODE_OUTPUT_PP_LOW);
+    gpio_init(SX_RX_EN, IO_MODE_OUTPUT_PP_LOW);
+    gpio_init(SX_RESET, IO_MODE_OUTPUT_PP_LOW);
+
+    gpio_init(SX2_DIO1, IO_MODE_INPUT_ANALOG);
+    gpio_init(SX2_BUSY, IO_MODE_INPUT_PU);
+    gpio_init(SX2_CS_IO, IO_MODE_OUTPUT_PP_HIGH);
+    gpio_init(SX2_TX_EN, IO_MODE_OUTPUT_PP_LOW);
+    gpio_init(SX2_RX_EN, IO_MODE_OUTPUT_PP_LOW);
+    gpio_init(SX2_RESET, IO_MODE_OUTPUT_PP_LOW);
+}
+
+bool sx_busy_read(void)
+{
+    return (digitalRead(SX_BUSY) == HIGH) ? true : false;
+}
+
+void sx_amp_transmit(void)
+{
+    gpio_low(SX_RX_EN);
+    gpio_high(SX_TX_EN);
+}
+
+void sx_amp_receive(void)
+{
+    gpio_low(SX_TX_EN);
+    gpio_high(SX_RX_EN);
+}
+
+void sx_dio_enable_exti_isr(void)
+{
+    attachInterrupt(SX_DIO1, SX_DIO_EXTI_IRQHandler, RISING);
+}
+
+void sx_dio_init_exti_isroff(void)
+{
+    detachInterrupt(SX_DIO1);
+}
+
+void sx_dio_exti_isr_clearflag(void) {}
+
+
+//-- Button
+
+#define BUTTON                    0
+
+void button_init(void)
+{
+    pinMode(BUTTON, INPUT_PULLUP);
+}
+
+bool button_pressed(void)
+{
+    return (digitalRead(BUTTON) == HIGH) ? false : true;
```

Comment:
there is a gpio_read_xxx() for this, see other hal

Reply:
Addressed